pkg/skaffold/initializer/analyze/builder.go

```diff
@@ -86,6 +88,14 @@ func (a *builderAnalyzer) detectBuilders(ctx context.Context, path string, detec
 		}
 	}
 
+	if a.enableKoInit {
+		if koinit.Validate(path) {
+			results = append(results, koinit.ArtifactConfig{
+				File: path,
+			})
+		}
+	}
+
```

Comment:
Can you potentially have multiple `go.mod` nested inside subdirectories?
If not, code should return `searchSubDirectories = false`

Reply:
> Can you potentially have multiple `go.mod` nested inside subdirectories?

Yes, and that should result in multiple artifacts.